mods/moontest/init.lua

```diff
@@ -24,24 +24,30 @@ minetest.settings:set_bool("smooth_lighting", true)
 minetest.register_item(":", { type = "none", wield_image = "hand.png"})
 skybox.add({"Space", "#FFFFFF", 0, { density = 0}})
 
-dofile(minetest.get_modpath("moontest") .. DIR_DELIM .. "src" .. DIR_DELIM .. "nodes.lua")
-dofile(minetest.get_modpath("moontest") .. DIR_DELIM .. "src" .. DIR_DELIM .. "habitat.lua")
-dofile(minetest.get_modpath("moontest") .. DIR_DELIM .. "src" .. DIR_DELIM .. "oxygen.lua")
-dofile(minetest.get_modpath("moontest") .. DIR_DELIM .. "src" .. DIR_DELIM .. "climate.lua")
-dofile(minetest.get_modpath("moontest") .. DIR_DELIM .. "src" .. DIR_DELIM .. "hunger.lua")
-dofile(minetest.get_modpath("moontest") .. DIR_DELIM .. "src" .. DIR_DELIM .. "energy.lua")
-dofile(minetest.get_modpath("moontest") .. DIR_DELIM .. "src" .. DIR_DELIM .. "machines.lua")
-dofile(minetest.get_modpath("moontest") .. DIR_DELIM .. "src" .. DIR_DELIM .. "interaction.lua")
-dofile(minetest.get_modpath("moontest") .. DIR_DELIM .. "src" .. DIR_DELIM .. "simulation.lua")
-dofile(minetest.get_modpath("moontest") .. DIR_DELIM .. "src" .. DIR_DELIM .. "reactor_booster.lua")
-dofile(minetest.get_modpath("moontest") .. DIR_DELIM .. "src" .. DIR_DELIM .. "logic.lua")
-dofile(minetest.get_modpath("moontest") .. DIR_DELIM .. "src" .. DIR_DELIM .. "aliens.lua")
-dofile(minetest.get_modpath("moontest") .. DIR_DELIM .. "src" .. DIR_DELIM .. "research.lua")
-dofile(minetest.get_modpath("moontest") .. DIR_DELIM .. "src" .. DIR_DELIM .. "sprint.lua")
-dofile(minetest.get_modpath("moontest") .. DIR_DELIM .. "src" .. DIR_DELIM .. "hud.lua")
-dofile(minetest.get_modpath("moontest") .. DIR_DELIM .. "src" .. DIR_DELIM .. "formspec.lua")
-dofile(minetest.get_modpath("moontest") .. DIR_DELIM .. "src" .. DIR_DELIM .. "tutorial.lua")
-dofile(minetest.get_modpath("moontest") .. DIR_DELIM .. "src" .. DIR_DELIM .. "shop_formspec.lua")
+-- Executes source files, just give .lua filename from src directory
+local do_src = function(filename)
```

Comment:
I make this function local so it doesn't conflict with any future mods that could be added to this game.

> This allows for just passing the filename and it executing that file from source directory. (Making any changes to directory structure easy to change, since it's in one spot)

Reply:
I like how the file name is passed to the function so the long form is only used once.

Not knowing about this pull request at the time, I moved all the do_file stuff into a file called do_file.lua and now just call do_file on that file and it calls it for the rest.
(https://github.com/Droog71/moontest/blob/main/mods/moontest/src/do_file.lua)

Maybe the do_src function could be moved to do_file.lua. 

I thought about doing something like this:

```
local path = minetest.get_modpath("moontest") .. "/src/"
local pfile = io.popen('ls -a "'..path..'"')
for filename in pfile:lines() do
    do_file(filename)
end
pfile:close()
```

Unfortunately, this would require adding the mod to the trusted mods list.

Reply:
Yup, just added it instead it's now do_file rather than do_src.